server/src/test/java/org/elasticsearch/index/mapper/TimeSeriesIdFieldMapperTests.java

```diff
@@ -653,4 +660,69 @@ public void testFewerDimensions() throws IOException {
         ParsedDocument doc2 = parseDocument(docMapper, d -> d.field("a", a).field("b", b).field("c", c));
         assertThat(doc1.rootDoc().getBinaryValue("_tsid").bytes, not(doc2.rootDoc().getBinaryValue("_tsid").bytes));
     }
+
+    public void testParseWithDynamicMapping() {
+        Settings indexSettings = Settings.builder()
+            .put(IndexSettings.MODE.getKey(), "time_series")
+            .put(IndexMetadata.INDEX_ROUTING_PATH.getKey(), "dim")
+            .build();
+        // without _id
+        {
+            MapperService mapper = createMapperService(IndexVersion.current(), indexSettings, () -> false);
+            SourceToParse source = new SourceToParse(null, new BytesArray("""
+                {
+                    "@timestamp": 1609459200000,
+                    "dim": "6a841a21",
+                    "value": 100
+                }"""), XContentType.JSON);
+            Engine.Index index = IndexShard.prepareIndex(
+                mapper,
+                source,
+                UNASSIGNED_SEQ_NO,
+                randomNonNegativeLong(),
+                Versions.MATCH_ANY,
+                VersionType.INTERNAL,
+                Engine.Operation.Origin.PRIMARY,
+                -1,
+                false,
+                UNASSIGNED_SEQ_NO,
+                0,
+                System.nanoTime()
+            );
+            assertNotNull(index.parsedDoc().dynamicMappingsUpdate());
+        }
+        // with _id
+        {
+            MapperService mapper = createMapperService(IndexVersion.current(), indexSettings, () -> false);
+            SourceToParse source = new SourceToParse("itaMgKqjeKVKQJobAAABdrs-cAA", new BytesArray("""
```

Comment:
Nit: consider setting the id to something more readable like "noSuchTsdbId"

Reply:
++, fixed in https://github.com/elastic/elasticsearch/pull/105636/commits/fc667e7b0e311cbce7073a9a1309f6919cfdf39a